tests/bluetooth/bsim_bt/bsim_test_mesh/src/settings_test_backend.c

```diff
@@ -21,6 +21,7 @@
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #define SETTINGS_FILE "settings_data.log"
```

Comment:
There is one thing about this (it was introduced in #36841 , this PR just builds on top)
These files are created in the bsim/bin/ folder directly. (That in itself is a bit naughty, as it leaves funny files around)
But, the main issues would be that any other simulation run at the same time using this settings_test_backend.c will also hit this same file independently of sim_id or device number. Is that intended? 
Or would you want to have this file created in the results/<sim_id>/ folder?
If so, https://github.com/BabbleSim/base/blob/master/libUtilv1/src/bs_results.c#L47 can create for you a result file like results/<sim_id>/d_<device_number>.<postfix>.csv
(as you also delete and rename files later, you'd probably prefer using `bs_create_result_folder` first to get the path to the results folder so you can find it after)
(To get the sim_id and device number you can do something like this:
```diff
diff --git a/boards/posix/nrf52_bsim/argparse.c b/boards/posix/nrf52_bsim/argparse.c
index 4b758186e9..2b0ab2e275 100644
--- a/boards/posix/nrf52_bsim/argparse.c
+++ b/boards/posix/nrf52_bsim/argparse.c
@@ -226,5 +226,16 @@ void nrfbsim_cleanup_args(void)
        bs_cleanup_dynargs(&args_struct);
 }
 
+char *get_simid(void)
+{
+       return arg.s_id;
+}
+
+unsigned int get_device_nbr(void)
+{
+       return arg.global_device_nbr;
+}
+
+
 NATIVE_TASK(nrfbsim_register_args, PRE_BOOT_1, 0);
 NATIVE_TASK(nrfbsim_cleanup_args, ON_EXIT, 10);
diff --git a/boards/posix/nrf52_bsim/argparse.h b/boards/posix/nrf52_bsim/argparse.h
index e52b4c9867..a7bae94138 100644
--- a/boards/posix/nrf52_bsim/argparse.h
+++ b/boards/posix/nrf52_bsim/argparse.h
@@ -26,6 +26,8 @@ struct NRF_bsim_args_t {
 
 struct NRF_bsim_args_t *nrfbsim_argsparse(int argc, char *argv[]);
 void bs_add_extra_dynargs(bs_args_struct_t *args_struct_toadd);
+char *get_simid(void);
+unsigned int get_device_nbr(void);
 
 #ifdef __cplusplus
 }
```
)

Reply:
@aescolar , I solved this issue in my PR (https://github.com/zephyrproject-rtos/zephyr/pull/37667). Please take a look.
https://github.com/zephyrproject-rtos/zephyr/pull/37667/files#diff-314130c4e23e5bb68050c36b5ff7b1749c8a4f17809b05dc00feef6378e815c8R85-R86
https://github.com/zephyrproject-rtos/zephyr/pull/37667/files#diff-f507577d8d0c7a424d5d0d418c82beea0d2aa006d2992f602a68de97b5e94731R211-R216

Probably it is worth to apply your suggestion. 
Generally, it is not possible to run two virtual devices with the permanent file name. They compete for the file.

Reply:
We agreed with @PavelVPV IRL to do changes discussed in this topic in my PR since they impact on my test and don't impact Pavel's test. @aescolar do you have objections?

Reply:
That's very fair